hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/Datas.java

```diff
@@ -0,0 +1,145 @@
+         * @throws EntityNotFoundException if there is no structured data on the current position in the inventory
+         * @see ResolvableToSingle#entity()
+         */
+        @Override
+        DataEntity entity() throws EntityNotFoundException;
+
+        /**
+         * Returns the data on the path relative to the entity.
+         *
+         * In another words, you can use this method to obtain only a subset of the data stored on the data entity.
+         *
+         * <p>I.e if you have an data entity which contains a map with a key "foo", which contains a list and you
+         * want to obtain a third element of that list, you'd do:
+         * {@code
+         * ...data(RelativePath.to().structuredData().key("foo").index(2).get());
+         * }
+         *
+         * <p>If you want to obtain the whole data structure, use an empty path: {@code RelativePath.empty().get()}.
+         *
+         * @param dataPath the path to the subset of the data.
+         * @return the subset of the data stored with the data entity
+         * @see #bareData(RelativePath)
+         */
+        StructuredData data(RelativePath dataPath);
+
+        /**
+         * This is very similar to {@link #data(RelativePath)} but this method doesn't load the child data.
+         *
+         * <p>If the data on the path contains a "primitive" value, the value is loaded. If the data contains a list,
+         * the returned instance will contain an empty list and if the data contains a map the returned instance will
+         * contain an empty map.
+         *
+         * @param dataPath the path to the subset of the data to return
+         * @return the subset of the data stored with the data entity
+         */
+        StructuredData bareData(RelativePath dataPath);
+
+        @Override
+        default boolean exists() {
+            try {
+                bareData(RelativePath.empty().get());
+                return true;
+            } catch (EntityNotFoundException | RelationNotFoundException ignored) {
+                return false;
+            }
+        }
+    }
+
+    public interface Multiple extends ResolvableToMany<DataEntity> {
+
+        /**
+         * Note that this is potentially expensive operation because it loads all the data associated with each of the
+         * returned data entities.
+         *
+         * @param pager the pager object describing the subset of the entities to return
+         * @return the page of the results
+         */
+        @Override
+        Page<DataEntity> entities(Pager pager);
+
+        /**
+         * Similar to {@link Single#data(RelativePath)}, only resolved over multiple entities.
+         *
+         * @param dataPath the path to the data entry inside the data entity
+         * @param pager    pager to use to page the data
+         * @return the page of the data entries
+         */
+        Page<StructuredData> datas(RelativePath dataPath, Pager pager);
```

Comment:
Well, config as such is not a single "datum" - it's data already ;) I don't like this myself either, but hope it gets the point across.